go/libraries/doltcore/schema/index_coll.go

```diff
@@ -17,16 +17,17 @@ package schema
 import (
 	"fmt"
 	"sort"
+	"strings"
 )
 
 type IndexCollection interface {
```

Comment:
Speaking more generally, about the implementation as a whole, I'm a little troubled by how tightly "hidden" is coupled to foreign keys. I think some of this coupling is necessary, and should be made explicit rather than abstracted behind a generic "hidden" flag. E.g. consider what happens when you create then drop a foreign key constraint under two scenarios: 1) there is a pre-existing index on the referenced columns, or 2) there isn't one. In the case of 1), we don't remove the index when we drop the constraint. In the case of 2) we do. So these indexes do need to have some knowledge of what they're being used for. Hidden is just too general.

Several ways this could be resolved, start thinking about it. Looking forward to seeing a solution in a future PR.